### PR TITLE
Cleaned up /datum/effect/system paths & some bad path defines

### PR DESCRIPTION
**Pull request by Tastyfish**
This seems like a lot of changes for the sake of refactoring, but the effect datums were a complete mess with the classes spread across.

##### Before
1. `/datum/effect/system`
 * Had no code or vars itself.
 * Only contained a couple explosion effects, both of which reinvented the wheel.
* `/datum/effect/effect/system`
 * This contained all other effects, despite the horrible path. `/datum/effect` is an empty superclass and `/datum/effect/effect` contains one proc, and the only subclass was `/datum/effect/effect/system`.
* `/datum/effects/system`.
 * Had no code or vars itself.
 * Contained a `sparkel` (sic) effect, which I fixed the spelling of, which also reinvented the wheel.

##### After
All of them are in `/datum/effect/system` and redundant vars removed. In a couple cases, redundant proc defintions were removed when identical.

#### Couple of bad parent classes
* There were also a couple of objs in `/obj/effects` instead of `/obj/effect`.

#### Examine Tab Descriptions
* There were 8 instances of the examine module defining descriptions for objects that didn't exist (The paths were wrong). This created 8 phantom objs that had no functionality polluting the object tree.

### CODE REVIEW

**Comment by Tastyfish** [comment]
I apologize in advance to the maintainer that has to look through this.

**Comment by Tastyfish** [comment]
Haha the actual metastation map was using one of the phantom examine objects, one moment.

File: code/game/objects/effects/effect_system.dm [hunk 1]
```diff
@@ -1198,17 +1198,17 @@ steam.start() -- spawns the effect
 
 
 //////////////////////////////////
-//////SPARKELS FIREWORKS
+//////SPARKLE FIREWORKS
 /////////////////////////////////
 ////////////////////////////
-/obj/effects/sparkels
-	name = "sparkel"
```
Comment (tigercat2000) [inline code/game/objects/effects/effect_system.dm]: [S](http://en.bab.la/dictionary/norwegian-english/sparkel)[p](https://forge.open-do.org/plugins/moinmoin/sparkel/FrontPage)[a](http://www.sparkel.net/)[r](http://www.sparkel.ca/)[k](http://www.sparkelgroup.nl/en/)[e](http://www.mrsparkel.co.nz/)[l](http://giphy.com/search/sparkel)[?](http://www.urbandictionary.com/define.php?term=sparkel)

**Comment by Tastyfish** [comment]
Yes, it is all of those things.

File: code/game/objects/effects/effect_system.dm (hunk 1 above)
Comment (Tastyfish) [inline code/game/objects/effects/effect_system.dm]: Yes, all of those things. (It's an effect for one of the fireworks, if you're curious.)

**Comment by TheDZD** [comment]
:cry:

**Comment by Fox-McCloud** [comment]
Who seriously thought this was a good idea?

:+1:

**Comment by Fox-McCloud** [comment]
Looks good to me.

@TheDZD @Markolie

**Comment by TheDZD** [comment]
Merge conflicts.

**Comment by Tastyfish** [comment]
conflict was from #2647 in code/modules/examine/descriptions/atmospherics.dm

**Comment by TheDZD** [comment]
Looks good otherwise, ready to be merged then?

**Comment by Tastyfish** [comment]
I believe so.

**Pull request merged**